File: CSharp/Services/SingletonState.cs
namespace CSharpDi.Services
{
    public class SingletonState
    {
        private int value;

        public SingletonState()
        {
            value = Random.Shared.Next(0, 100);
        }

        public int GetValue()
        {
            return value;
        }
    }
}

File: TypeScript/services/SingletonState.ts
import { Lifecycle, scoped } from "tsyringe";

@scoped(Lifecycle.ContainerScoped)
export class SingletonState {
    value: number;

    constructor() {
        this.value = Math.floor(Math.random() * 100);
    }

    getValue(): number {
        return this.value;
    }
}